src/osd/ReplicatedPG.cc

```diff
@@ -5774,7 +5774,7 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
       ctx->mod_desc.mark_unrollbackable();
       ++ctx->num_write;
       {
-	if (!obs.exists || oi.is_whiteout()) {
+	if (!obs.exists || oi.is_whiteout() || !oi.is_omap()) {
```

Comment:
if the object does not have any omap entry, i think we can consider this op done, and "break" right away instead of returning a '-ENOENT'?